src/stirling/source_connectors/socket_tracer/bcc_bpf_intf/socket_trace.hpp

```diff
@@ -134,28 +146,51 @@ struct SocketDataEvent {
 
     DCHECK_GE(attr.msg_size, attr.msg_buf_size);
 
-    if (attr.msg_size > attr.msg_buf_size) {
-      VLOG(1) << "Adding filler to event";
+    // Note that msg_size - msg_buf_size != bytes_missed in the case where we exceed LOOP_LIMIT
+    // in perf_submit_iovecs, because one call to perf_submit_buf takes only the size of the current
+    // iovec into account, ommitting the rest of the iovecs which could not be submitted.
+    // As a result, we need to use bytes_missed to determine the size of the filler event.
+
+    // For kernels < 5.1, we cannot track the bytes missed in socket_trace.c properly and thus
+    // preserve the previous behavior of encoding the bytes missed via the msg_size.
+    // If our loop and chunk limits are at most 42 and 4, then we know that we can
+    // stay below the verifier instruction limit for kernels < 5.1.
+    if (LOOP_LIMIT <= 42 && CHUNK_LIMIT <= 4) {
```

Comment:
Please consider here and in the eBPF c code, creating a semantically meaningful version of this. Perhaps:
```
const bool kernelNewerThan5dot1 = LOOP_LIMIT > 42 || CUNK_LIMIT > 4;
```

Reply:
Added